pkgs/by-name/ko/koboldcpp/package.nix

```diff
@@ -106,14 +119,31 @@ effectiveStdenv.mkDerivation (finalAttrs: {
   env.NIX_LDFLAGS = lib.concatStringsSep " " (finalAttrs.darwinLdFlags ++ finalAttrs.metalLdFlags);
 
   env.NIX_CFLAGS_COMPILE =
-    lib.optionalString (march != "") "-march=${march}" + lib.optionalString (mtune != "") "-mtune=${mtune}";
+    let
+      inherit (stdenv.hostPlatform)
+        isx86_64
+        isAarch64
+        isLinux
+        isDarwin
+        ;
+    in
+    lib.concatStringsSep " " [
+      (lib.optionalString isx86_64 "-march=x86-64")
+      (lib.optionalString (isAarch64 && isLinux) "-march=armv8.7-a")
```

Comment:
This one does specify and extended instruction set which will be different from what the user has "requested" with their Nixpkgs configuration (the `config` and `system` they passed to `import <nixpkgs>`). I'm not saying this is necessarily wrong, but at least it's not obvious why it's this way